OpenGL/Math/Matrix2.cs

```diff
@@ -86,6 +86,44 @@ public static Matrix2 Identity
             }
         }
 
+        public float this[int column, int row]
+        {
+            get
+            {                    
+                if (row == 0)
+                {
+                    if (column == 0) return row1.X;
+                    if (column == 1) return row1.Y;
+                }
+                else if (row == 1)
+                {
+                    if (column == 0) return row2.X;
+                    if (column == 1) return row2.Y;
+                }
+                else
+                {
+                    throw new ArgumentOutOfRangeException("column or row is out of the matrix bounds.");                    
+                }
+                return -1; //for some reason, it needs this, or get will bomb.
```

Comment:
If you remove it, there is a red error line over `get`. I don't understand why though.

Reply:
The reason why is because you aren't throwing an exception if the column is out of range.  Something like this should work:

```
get
{       
    if (column < 0 || column > 2) throw new ArgumentOutOfRangeException("column");
    if (row < 0 || row > 2) throw new ArgumentOutOfRangeException("row");
    else if (row == 0)
    {
        if (column == 0) return row1.X;
        else return row1.Y;
    }
    else
    {
        if (column == 0) return row2.X;
        else return row2.Y;
    }
}
```

We could extend `Vector2` to support `Get` and even throw the Exception in there, making this code even simpler (just like what was done for `Matrix3`).